metrics/firebase/firestore/rules/firestore.rules

```diff
@@ -0,0 +1,142 @@
+rules_version = '2';
+service cloud.firestore {
+  match /databases/{database}/documents {
+    
+    match /projects/{projectId} {
+      allow read: if isLoggedIn();
+      allow create, update: if isLoggedIn() && isProjectValid();
+      allow delete: if false;
+    }
+
+
+    match /build/{buildId} {
+      allow read: if isLoggedIn();
+      allow create, update: if isLoggedIn() && isBuildValid(database);
+      allow delete: if false;
+    }
+  }
+
+  /// Checks if a request is from a logged-in user.
+  function isLoggedIn() {
+    return request.auth.uid != null
+  }
+
+  /// Checks whether project data from the request is valid.
+  function isProjectValid(){
+    let requestData = request.resource.data;
+
+    return isValidProjectData(requestData) && isString("name");
+  }
+
+  /// Check whether the data contains only allowed for project document fields.
+  function isValidProjectData(data){
+    return data.keys().hasOnly(["name"]);
+  }
+
+  /// Checks whether the field is the string.
+  function isString(field){
+    let requestData = request.resource.data;
+
+    return field in requestData && requestData[field] is string;
+  }
+
+  /// Checks whether build data from the request is valid.
+  function isBuildValid(database){
+    let requestData = request.resource.data;
+
+    return isValidBuildData(requestData)
+          && projectExists(database)
+          && isIntegerOrNull("buildNumber")
```

Comment:
```suggestion
          && isIntegerOrNull("buildNumber")
```
please see if we can add optional parameter or overload method to add additional parameter - "nullable"